docs: create "add a linked ssh key", "deploy ssh key" articles and modify "link ssh key" (#759)
devopness/devopness: docs/ssh-keys/deploy-ssh-key.md

---
title: Deploy SSH Key

> intro: Deploy your SSH Keys to each of the servers you want the keys on.

Reviewer: ```suggestion
intro: Deploy your SSH Keys to each of the servers you want to give access to using those keys.
```

I believe with this change it is more clear on what is the goal of deploying a ssh key to a server: to give access to a server.

Do you agree?

links:
    overview:
    quickstart:
    previous:
        - /docs/ssh-keys/link-ssh-key
    next:
        - /docs/servers/ssh-into-server
    guides:
    related:
    featured:
required_permissions:

>     - application:deploy

Reviewer: ```suggestion
    - ssh-key:pipeline_run
```

SSH Key does not require application permissions; and instead of deploy, the API requires the user to be able to run pipelines.

Could you update this section?

>     - application:pipeline_run

Reviewer 2: This permission has been renamed to `application:deploy`.

```suggestion
    - application:deploy
```

---

1. On Devopness, navigate to a project then select an environment
1. Find the `SSH Keys` card
1. Click `View` in the `SSH Keys` card to see a list of existing `SSH Keys`
1. Click `DEPLOY` on the SSH Key you want to deploy

> 1. Choose the pipeline to be deployed, then click `Next`

Author: Both sentences look semantically equal to me. In "Choose the pipeline to be deployed, then click `Next`", "the" does not mean a singular, specific pipeline.

> 1. Choose the pipeline to be deployed, then click `Next`

Reviewer: @brlipi yes, because although the first one is semantically correct, I believe the second one uses certain words to help guide the user to success.

> "Choose the pipeline to be deployed"

In this sentence, I understand that any pipeline I choose to deploy[1] will lead me to the article expected output: deploy an ssh key to the server.

> "Choose a deploy pipeline to be run"

In this case there's a highlight that it needs to be a "deploy pipeline".

[1] I know the current `DEPLOY` button may lead to some confusion, but we _deploy_ resources by _running_ pipelines, _deploy_ is a pipeline operation; in the docs, we try helping our users achieve success by using the same naming patterns as the current app version, making it easier to recognize.

> 1. Choose the pipeline to be deployed, then click `Next`

Author: Understood.

> 1. Choose the branch and the pipeline to be deployed, then click `Next`

Reviewer 2: "Choose the branch". I think this description have been copied from another doc, right? Because SSH doesn't have "branch".

```suggestion
1. Choose the pipeline to be deployed, then click `Next`
```

1. Review deploy information then click `Confirm`

>    > A notification popup will be displayed, confirming that the deployment has been triggered

Reviewer: ```suggestion
```

I tested the form a couple of times and did not see any notification pop up on the Run Pipeline Form.

It is a nice to have feature, but currently not implemented.

For now, could you remove this note?
